Src/Support/Google.Apis.Auth.Tests/OAuth2/GoogleCredentialTests.cs

```diff
@@ -71,6 +82,79 @@ public class GoogleCredentialTests
+GS37bwJANdSAKfbafLfjuhqwUJ9yGpykZm/a36aTmerp/bpn1iHdg+RtCzwMcDb/
+TWSwibbvsflgWmHbz657y4WSWhq+8QJAWrpCNN/ZCk2zuGDo80lfUBAwkoVat8G6
+wWU1oZyS+vzIGef+hLb8kHsjeZPej9eIwZ39kcBbT54oELrCkRjwGwJAQ8V2A7lT
+ZUp8AsbVqF6rbLiiUfJMo2btGclQu4DEVyS+ymFA65tXDLUuR9EDqJYdqHNZJ5B8
+4Z5p2prkjWTLcA\u003d\u003d
+-----END PRIVATE KEY-----"",
+""ClientEmail"": ""CLIENT_EMAIL"",
+""ClientId"": ""CLIENT_ID"",
+""ProjectId"": ""PROJECT_ID"",
+""Type"": ""service_account""}}";
+
+        private IConfiguration BuildConfigurationForCredential(string json) =>
+            // We parse the string and call ToString after because within the key
+            // there's character that need scaping. Newtonsoft seems to handle them
+            // by default, whereas System.Text.Json, which is what Microsoft.Extensions.Configuration
+            // depends on, does not.
+            // Note that this has no direct effect on Google.Apis.Auth, as, at most, this affects
+            // the step of writing the service account key to configuration, which is external to
+            // this library.
+            new ConfigurationBuilder().AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(JToken.Parse(json).ToString()))).Build();
+
+        [Fact]
+        public void FromConfig_UserCredential()
+        {
+            IConfiguration config = BuildConfigurationForCredential(DummyUserCredentialConfigContents);
+            JsonCredentialParameters credentialParameters = config.GetSection("GoogleCredential").Get<JsonCredentialParameters>();
+            GoogleCredential credential = GoogleCredential.FromJsonParameters(credentialParameters);
+
+            var userCred = Assert.IsType<UserCredential>(credential.UnderlyingCredential);
+            Assert.False(credential.IsCreateScopedRequired);
+            Assert.Equal("REFRESH_TOKEN", userCred.Token.RefreshToken);
+            var flow = (GoogleAuthorizationCodeFlow)userCred.Flow;
+            Assert.Equal("CLIENT_ID", flow.ClientSecrets.ClientId);
+            Assert.Equal("CLIENT_SECRET", flow.ClientSecrets.ClientSecret);
+            Assert.Equal("PROJECT_ID", flow.ProjectId);
+            Assert.Equal("QUOTA_PROJECT", userCred.QuotaProject);
+        }
+
+        [Fact]
+        public void FromConfig_ServiceAccountCredential()
+        {
+            IConfiguration config = BuildConfigurationForCredential(DummyServiceAccountCredentialConfigContents);
+            JsonCredentialParameters credentialParameters = config.GetSection("GoogleCredential").Get<JsonCredentialParameters>();
+            GoogleCredential credential = GoogleCredential.FromJsonParameters(credentialParameters);
+
+            var serviceCred = Assert.IsType<ServiceAccountCredential>(credential.UnderlyingCredential);
+            Assert.True(credential.IsCreateScopedRequired);
+            Assert.Equal("CLIENT_EMAIL", serviceCred.Id);
+            Assert.Equal("PROJECT_ID", serviceCred.ProjectId);
+        }
+
```

Comment:
Add tests for a) invalid JSON; b) valid JSON that doesn't contain everything we expect; c) valid JSON that doesn't conform with the types in our class? (I'd expect these to be pretty simple, but they'd document the expected behavior.)

Reply:
Tests added.